pandas/io/excel/_base.py

```diff
@@ -844,14 +844,24 @@ class ExcelFile:
 
     def __init__(self, path_or_buffer, engine=None):
         if engine is None:
-            engine = "xlrd"
+            engine = "openpyxl"
```

Comment:
So this actually changes the engine; I think the first step is to provide the warning that you have below that by default in the future we will switch to using openpyxl

Reply:
Making an exception for .xls files which have to use xlrd

Reply:
So that means that a warning shall only be produced for .xlsx and .xlsm files that use xlrd?
And regarding the other remark about switching the engines, that was what you asked for a couple of comments back?

Reply:
That's right - warn first then change over time

Reply:
so the change to use the openpyxl engine as the default has to be reverted?

Reply:
@WillAyd Should I revert the change to make openpyxl the default engine?
And only warn when using xlrd in combination with .xlsx or .xlsm files?